### ISSUE TITLE
DINO with more than 6 layers

### ISSUE THREAD

**Issue by R2Bb1T** [open-mmlab/mmdetection#11568]
I am training DINO on my own dataset of enc/dec layers of 4/4, 5/5, 6/6, 4/6 and all the result seems reasonable.
But when it comes to 7/7 or 8/8, the evaluation results go wrong. And I tried to set the encoder config `num_cp=7`, but it still doesn't work.
the model config is below:
`model = dict(
    as_two_stage=True,
    backbone=dict(
        depth=50,
        frozen_stages=1,
        init_cfg=dict(checkpoint='torchvision://resnet50', type='Pretrained'),
        norm_cfg=dict(requires_grad=False, type='BN'),
        norm_eval=True,
        num_stages=4,
        out_indices=(
            0,
            1,
            2,
            3,
        ),
        style='pytorch',
        type='ResNet',
        with_cp=True),
    bbox_head=dict(
        loss_bbox=dict(loss_weight=5.0, type='L1Loss'),
        loss_cls=dict(
            alpha=0.25,
            gamma=2.0,
            loss_weight=1.0,
            type='FocalLoss',
            use_sigmoid=True),
        loss_iou=dict(loss_weight=2.0, type='GIoULoss'),
        num_classes=1,
        sync_cls_avg_factor=True,
        type='DINOHead'),
    data_preprocessor=dict(
        bgr_to_rgb=True,
        mean=[
            123.675,
            116.28,
            103.53,
        ],
        pad_size_divisor=1,
        std=[
            58.395,
            57.12,
            57.375,
        ],
        type='DetDataPreprocessor'),
    decoder=dict(
        layer_cfg=dict(
            cross_attn_cfg=dict(dropout=0.0, embed_dims=256, num_levels=5),
            ffn_cfg=dict(
                embed_dims=256, feedforward_channels=2048, ffn_drop=0.0),
            self_attn_cfg=dict(dropout=0.0, embed_dims=256, num_heads=8)),
        num_layers=7,
        post_norm_cfg=None,
        return_intermediate=True),
    dn_cfg=dict(
        box_noise_scale=1.0,
        group_cfg=dict(dynamic=True, num_dn_queries=150, num_groups=None),
        label_noise_scale=0.5),
    encoder=dict(
        layer_cfg=dict(
            ffn_cfg=dict(
                embed_dims=256, feedforward_channels=2048, ffn_drop=0.0),
            self_attn_cfg=dict(dropout=0.0, embed_dims=256, num_levels=5)),
        num_cp=7,
        num_layers=7),
    neck=dict(
        act_cfg=None,
        in_channels=[
            256,
            512,
            1024,
            2048,
        ],
        kernel_size=1,
        norm_cfg=dict(num_groups=32, type='GN'),
        num_outs=5,
        out_channels=256,
        type='ChannelMapper'),
    num_feature_levels=5,
    num_queries=1500,
    positional_encoding=dict(
        normalize=True, num_feats=128, offset=0.0, temperature=20),
    test_cfg=dict(max_per_img=512),
    train_cfg=dict(
        assigner=dict(
            match_costs=[
                dict(type='FocalLossCost', weight=2.0),
                dict(box_format='xywh', type='BBoxL1Cost', weight=5.0),
                dict(iou_mode='giou', type='IoUCost', weight=2.0),
            ],
            type='HungarianAssigner')),
    type='DINO',
    with_box_refine=True)`
the loss began with 55.6 and ended with 21.9307, but the evaluation result is `bbox_mAP_copypaste: 0.00170 0.028 0.000 0.001 0.006 0.005 0.010 0.031 0.050`
The loss of normal model ends with around 3, is the layers larger than 6 automatically frozen?

**Comment by Chop1** [comment]
Maybe you are using the pretrained model, which use 6 layers. If you lessen the layer, you are still using the trained weight. I you add extra layer, those are randomly initialized and need large scale dataset to be trained.

Its just a guess.

**Comment by R2Bb1T** [comment]
> Maybe you are using the pretrained model, which use 6 layers. If you lessen the layer, you are still using the trained weight. I you add extra layer, those are randomly initialized and need large scale dataset to be trained.
> 
> Its just a guess.

Thanks for replying! But I checked the log and it showed that onlly the backbone is loaded from pretrained model. And the init information of layer 7 and layers before are the same, which is `The value is the same before and after calling init_weights of DINO`, so I think they were all trained from scratch.
And I think when the model mismatches with the checkpoint, it will still load the matched parameters and init the rest?
